test/built-ins/TypedArray/prototype/slice/arraylength-internal.js

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.slice
+description: Use internal ArrayLength instead of getting a length property
+info: >
+  22.2.3.24 %TypedArray%.prototype.slice ( start, end )
+
+  ...
+  3. Let len be the value of O's [[ArrayLength]] internal slot.
+  ...
+includes: [testTypedArray.js, compareArray.js]
+---*/
+
+var getCalls = 0;
+var desc = {
+  get: function getLen() {
+    getCalls++;
+    return 0;
+  }
+};
+
+Object.defineProperty(TypedArray.prototype, "length", desc);
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([42, 43]);
+  getCalls = 0;
```

Comment:
It's not enough to re-set the counter here because the [[ArrayLength]] value is something you are trying to test.

As written, the call to `compareArray` will always return `true` because it operates as though it were given two array-likes with length `0`. For instance, you can insert this statement on line 33, and the test will continue to pass:

    result[0] = 9;